lib/dry/web/roda/templates/Rakefile.tt

```diff
@@ -15,11 +15,30 @@ def db
   <%= config[:camel_cased_app_name] %>::Container["persistence.db"]
 end
 
+def settings
+  <%= config[:camel_cased_app_name] %>::Container["settings"]
+end
+
+def execute_database_create_command(settings)
+  database_url = settings.database_url
+  _, _, _, db_name = database_url.split('/')
+  if user_and_net.include?('@')
```

Comment:
I can't see where `user_and_net` is assigned? Did you leave something out, perhaps? Anyway, perhaps it might be worth trying the URI.parse approach I mentioned above.